src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,243 @@
+const REQUEST_TIMEOUT = 8000; // ms
+const request = require('request-promise-native');
+const { Configuration } = require('@schul-cloud/commons');
+const { Forbidden, GeneralError } = require('@feathersjs/errors');
+const logger = require('../../../logger');
+
+
+class EduSharingConnector {
+	constructor() {
+		if (EduSharingConnector.instance) {
+			return EduSharingConnector.instance;
+		}
+		this.authorization = null; // JSESSION COOKIE
+		this.accessToken = null; // ACCESSTOKEN
+		EduSharingConnector.instance = this;
+	}
+
+	static get headers() {
+		return {
+			Accept: 'application/json',
+			'Content-type': 'application/json',
+		};
+	}
+
+	static get authorization() {
+		const headers = {
+			...EduSharingConnector.headers,
+			Authorization: `Basic ${Buffer.from(`${Configuration.get('ES_USER')
+			}:${Configuration.get('ES_PASSWORD')}`).toString(
+				'base64',
+			)}`,
+		};
+
+		return headers;
+	}
+
+
+	// gets cookie (JSESSION) and attach it to header
+	getCookie() {
+		const cookieOptions = {
+			uri: `${Configuration.get('ES_DOMAIN')}/edu-sharing/rest/authentication/v1/validateSession`,
+			method: 'GET',
+			headers: EduSharingConnector.authorization,
+			resolveWithFullResponse: true,
+			json: true,
+		};
+		return request(cookieOptions)
+			.then((result) => {
+				if (
+					result.statusCode !== 200
+					|| result.body.isValidLogin !== true
+				) {
+					throw Error('authentication error with edu sharing');
+				}
+				return result.headers['set-cookie'][0];
+			})
+			.catch((err) => {
+				// eslint-disable-next-line no-console
+				console.error('error: ', err);
+			});
+	}
+
+	// gets access_token and refresh_token
+	getAuth() {
+		const oauthoptions = {
+			method: 'POST',
+			url: `${Configuration.get('ES_DOMAIN')}/edu-sharing/oauth2/token`,
+			headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+
+			// eslint-disable-next-line max-len
+			body: `grant_type=${Configuration.get('ES_GRANT_TYPE')}&client_id=${
+				Configuration.get('ES_CLIENT_ID')
+			}&client_secret=${Configuration.get('ES_OAUTH_SECRET')}&username=${
+				Configuration.get('ES_USER')
+			}&password=${Configuration.get('ES_PASSWORD')}`,
+			timeout: REQUEST_TIMEOUT,
+		};
+		return request(oauthoptions).then((result) => {
+			if (result) {
+				const parsedResult = JSON.parse(result);
+				return parsedResult.access_token;
+			}
+			// eslint-disable-next-line no-console
+			console.error('Oauth failed');
+			return null;
+		});
+	}
+
+	allConfigurationValuesHaveBeenDefined() {
+		return (
+			Configuration.get('ES_DOMAIN')
```

Comment:
however use 
```suggestion
			Configuration.has('ES_DOMAIN')
```
for all values here to _not_ implictly take the truthy values

Reply:
Ok